solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2299,6 +2300,25 @@ public long getStatus() {
     return status;
   }
 
+  /**
+   * Retrieve the aliases from zookeeper. This is typically cached and does not hit zookeeper after
+   * the first use.
+   *
+   * @return an immutable instance of {@code Aliases} accurate as of at the time this method is
+   *     invoked, less any zookeeper update lag.
+   * @throws RuntimeException if invoked on a {@code CoreContainer} where {@link
+   *     #isZooKeeperAware()} returns false
+   */
+  public Aliases getAliases() {
+    if (isZooKeeperAware()) {
+      return getZkController().getZkStateReader().getAliases();
+    } else {
+      // fail fast because it's programmer error, but give slightly more info than NPE.
+      throw new IllegalStateException(
```

Comment:
When I was looking at this I felt that returning empty was likely to hide lurking misuse, and documenting this also makes things clearer for someone learning the code base. Both the javadoc and the code now make it clear that this is something only germane to cloud usage, whereas before one had to be familiar enough to realize that "isZookeeperAware()" is solr-code-speak for "isThisCloudAndNotUserManagedStandaloneOrWhateverWeCallItNow()" ;) Also, hopefully they can also now know that it's a safe/cheap thing to call  and doesn't pound on zookeeper without reading much more complicated code. Anyway that's what I was thinking. I could possibly be convinced to not backport this aspect to 8.11.2 however since in theory it might expose any existing bad code in a harmful way.

Reply:
Ok, I can get on board with throwing exceptions in 9.x and leaving it returning EMPTY in 8.x